minetest/minetest
# Use openssl's sha1 and sha256

* Goal: Make sha1 and sha256 computations faster. (The implementations in master don't use hardware extensions.)
  This mostly influences server startup and client join times. But lua api sha calls are also affected.
* OpenSSL3 is added as dependency (its libcrypto).
* The new hash util api is very simple: `hashing::sha1()` and `hashing::sha256()`. (Namespace `hashing`, because I wanted to avoid name conflicts.)
* `srp.cpp` uses openssl directly.

## To do

This PR is a Ready for Review, but not mergeable yet.

I need help for the CI stuff.

## How to test

* Start a singleplayer game, and look at your clock.
  ==> Wuzzy's Lazarr! game for example starts about twice as fast (less than 3 vs. about 5 seconds).
* You can also see the effects in a profiler.
